controllers/dspipeline_controller_test.go

```diff
@@ -130,12 +130,12 @@ func deleteDSP(path string, opts mf.Option) {
 	dsp2 := &dspav1alpha1.DataSciencePipelinesApplication{}
 	Eventually(func() error {
 		namespacedNamed := types.NamespacedName{Name: dsp.Name, Namespace: WorkingNamespace}
-		Expect(k8sClient.Get(ctx, namespacedNamed, dsp2)).NotTo(HaveOccurred())
-		if !reflect.DeepEqual(dsp2, &dspav1alpha1.DataSciencePipelinesApplication{}) {
-			return errors.New("DSP still exists on cluster")
+		err = k8sClient.Get(ctx, namespacedNamed, dsp2)
+		if err != nil && apierrs.IsNotFound(err) {
```

Comment:
this looks like we're doing some sort of 'unknown error' ignoring here.  Is that the case, and if so why?

Reply:
Can you explain? 

This makes sense when the whole code block is viewed: 

```golang
	Eventually(func() error {
		namespacedNamed := types.NamespacedName{Name: dsp.Name, Namespace: WorkingNamespace}
		err = k8sClient.Get(ctx, namespacedNamed, dsp2)
		if err != nil && apierrs.IsNotFound(err) {
			return nil
		}
		return fmt.Errorf("resource still exists on cluster")
	}, timeout, interval).ShouldNot(HaveOccurred())
```

This tests an async call, which is basically doing: 

Try to fetch this DSP resource, **we are expecting** it to not be on the cluster (IsNotFound). **We are not expecting** it to exist on the cluster, or encounter any other error while attempting to fetch.

Reply:
I see what you are saying, there's a missing case here, I'll update

Reply:
updated